gobblin-runtime/src/main/java/org/apache/gobblin/runtime/api/MysqlMultiActiveLeaseArbiter.java

```diff
@@ -359,10 +366,13 @@ else if (leaseValidityStatus == 2) {
    * Checks leaseArbiterTable for an existing entry for this flow action and event time
    */
   protected Optional<GetEventInfoResult> getExistingEventInfo(DagActionStore.DagAction flowAction,
-      boolean isReminderEvent) throws IOException {
+      boolean isReminderEvent, long eventTimeMillis) throws IOException {
     return withPreparedStatement(isReminderEvent ? thisTableGetInfoStatementForReminder : thisTableGetInfoStatement,
         getInfoStatement -> {
           int i = 0;
+          if (isReminderEvent) {
```

Comment:
Why not get the timestamp regardless if it's a reminder event or not? Seems useful

Reply:
This is using the timestamp for comparison to the db event timestamp only for reminder event. Otherwise for _original events _ we do db laundering and use current timestamp in db.